snappy/app/converge.go

```diff
@@ -257,8 +263,20 @@ func (h *Handler) snapToPayload(snapQ *client.Snap) snapPkg {
 	if isInstalled {
 		iconPath, err := localIconPath(h.snapdClient, snap.Name)
 		if err != nil {
-			log.Println("Icon path for installed package cannot be set", err)
-			iconPath = ""
+			if err == ErrIconNotExist {
+				// We have an installed snap, but no icon found,
+				// this could happen during an uninstall process.
+				var p string
```

Comment:
Feels like var p is unneeded : you could directly assign to iconPath and modify if err != nil ?